optimade/filtertransformers/mongo.py

```diff
@@ -86,3 +86,180 @@ def combined(self, args):
+        # TODO: Probably the value_op_rhs rule is not the best for implementing this.
+        return {prop: {self.reversed_operator_map[oper]: arg[0]} for oper, prop in arg[1].items()}
+
+    def predicate_comparison(self, arg):
+        # predicate_comparison: length_comparison
+        raise NotImplementedError
+
+    @v_args(inline=True)
+    def value_op_rhs(self, operator, value):
+        # value_op_rhs: OPERATOR value
+        return {self.operator_map[operator]: value}
+
+    def known_op_rhs(self, arg):
+        # known_op_rhs: IS ( KNOWN | UNKNOWN )
+        if arg[1] == 'KNOWN':
+            return {'$exists': True}
+        elif arg[1] == 'UNKNOWN':
+            return {'$exists': False}
+
+    def fuzzy_string_op_rhs(self, arg):
+        # fuzzy_string_op_rhs: CONTAINS string | STARTS [ WITH ] string | ENDS [ WITH ] string
+
+        # The WITH keyword may be omitted.
+        if isinstance(arg[1], Token) and arg[1].type == "WITH":
+            pattern = arg[2]
+        else:
+            pattern = arg[1]
+
+        if arg[0] == 'CONTAINS':
+            return {'$regex': f'{pattern}'}
+        elif arg[0] == 'STARTS':
+            return {'$regex': f'^{pattern}'}
+        elif arg[0] == 'ENDS':
+            return {'$regex': f'{pattern}$'}
+
+    def set_op_rhs(self, arg):
+        # set_op_rhs: HAS ( [ OPERATOR ] value | ALL value_list | ANY value_list | ONLY value_list )
+
+        if len(arg) == 2:
+            # only value without OPERATOR
+            return {'$in': arg[1]}
+        else:
+            if arg[1] == 'ALL':
+                raise NotImplementedError
+            elif arg[1] == 'ANY':
+                raise NotImplementedError
+            elif arg[1] == 'ONLY':
+                raise NotImplementedError
+            else:
+                # value with OPERATOR
+                raise NotImplementedError
+
+    def set_zip_op_rhs(self, arg):
+        # set_zip_op_rhs: property_zip_addon HAS ( value_zip | ONLY value_zip_list | ALL value_zip_list |
+        # ANY value_zip_list )
+        raise NotImplementedError
+
+    def length_comparison(self, arg):
+        # length_comparison: LENGTH property OPERATOR value
+        raise NotImplementedError
+
+    def property_zip_addon(self, arg):
+        # property_zip_addon: ":" property (":" property)*
+        raise NotImplementedError
+
+    def property(self, arg):
+        # property: IDENTIFIER ( "." IDENTIFIER )*
+        return '.'.join(arg)
+
+    @v_args(inline=True)
+    def string(self, string):
+        # string: ESCAPED_STRING
+        return string.strip('"')
+
+    def number(self, arg):
+        # number: SIGNED_INT | SIGNED_FLOAT
+        token = arg[0]
+        if token.type == 'SIGNED_INT':
+            return int(token)
+        elif token.type == 'SIGNED_FLOAT':
+            return float(token)
+
+    def __default__(self, data, children, meta):
+        raise NotImplementedError
+
+
+if __name__ == '__main__':  # pragma: no cover
```

Comment:
I am assuming this is your sort-of testing space? Would it be okay with you to leave this out (perhaps stash it or similar) or else create a proper interface to test the parser? Otherwise, i see it a bit as bloat that is convenient for developing the parser, but that's it.